rfremoterunner/executor.py

```diff
@@ -0,0 +1,58 @@
+import sys
+import os
+import logging
+from rfremoterunner.utils import write_file_to_disk
+from rfremoterunner.executor_argparser import ExecutorArgumentParser
+from rfremoterunner.rf_client import RemoteFrameworkClient
+
+logger = logging.getLogger('rfremoterunner.executor')
+out_hdlr = logging.StreamHandler(sys.stdout)
+out_hdlr.setFormatter(logging.Formatter('%(message)s'))
+out_hdlr.setLevel(logging.DEBUG)
+logger.addHandler(out_hdlr)
+logger.setLevel(logging.INFO)
+
+
+def run_executor():
+    """
+    Initialise and run the executor
+    """
+    # Parse the input arguments
+    arg_parser = ExecutorArgumentParser(sys.argv[1:])

```

Comment:
element 0 for sys.argv is the name of the script, so if there are no arguments specified then sys.argv[1:] evaluates to an empty list. This is then thrown into argparse which then raises an exception because of too few arguments:

usage: executor.py [-h] [--debug] [-d OUTPUTDIR] [-o OUTPUT] [-l LOG]
                   [-r REPORT] [-F EXTENSION] [-s SUITE] [-t TEST]
                   [-i INCLUDE] [-e EXCLUDE] [-L LOGLEVEL]
                   host suites [suites ...]
executor.py: error: too few arguments